### PR TITLE
check for debugging lint in CI

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 we forget println and eprintln while dev'ing and testing

 #### Summary of Changes
 catch us forgetting

### CODE REVIEW

File: ci/nits.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# please don't print from --lib...
+declare prints=(
+  'print!'
+  'println!'
+  'eprint!'
+  'eprintln!'
+)
+
+if _ git grep "${prints[@]/#/-e }" src
```
Comment (mvines) [inline ci/nits.sh]: nit for `nits.sh`:
```bash
if _ git grep "${prints[@]/#/-e }" src; then
```

Comment (mvines) [reply ci/nits.sh]: Also how about checking `*/src` in addition to `src`

File: ci/test-checks.sh [hunk 2]
```diff
@@ -13,6 +13,8 @@ _ cargo fmt --all -- --check
 _ cargo clippy --all -- --version
 _ cargo clippy --all -- --deny=warnings
 _ ci/audit.sh
+_ ci/nits.sh || true # catch println!
 _ book/build.sh
 
+
```
Comment (mvines) [inline ci/test-checks.sh]: nit: why extra newline

File: ci/nits.sh (hunk 1 above)
Comment (rob-solana) [reply ci/nits.sh]: most, I think.

**Pull request merged**